x-pack/plugins/infra/public/pages/metrics/inventory_view/components/layout.tsx

```diff
@@ -52,186 +51,173 @@ interface LegendControlOptions {
+    metric,
+    sort,
+    groupBy,
+  };
+
+  useInterval(
+    () => {
+      if (!loading) {
+        jumpToTime(Date.now());
       }
-    }, [currentView, onViewChange]);
-
-    useEffect(() => {
-      // load snapshot data after default view loaded, unless we're not loading a view
-      if (currentView != null || !shouldLoadDefault) {
-        reload();
-      }
-
-      /**
-       * INFO: why disable exhaustive-deps
-       * We need to wait on the currentView not to be null because it is loaded async and could change the view state.
-       * We don't actually need to watch the value of currentView though, since the view state will be synched up by the
-       * changing params in the reload method so we should only "watch" the reload method.
-       *
-       * TODO: Should refactor this in the future to make it more clear where all the view state is coming
-       * from and it's precedence [query params, localStorage, defaultView, out of the box view]
-       */
-      /* eslint-disable-next-line react-hooks/exhaustive-deps */
-    }, [reload, shouldLoadDefault]);
-
-    useEffect(() => {
-      setShowLoading(true);
-    }, [options.metric, nodeType]);
-
-    useEffect(() => {
-      const hasNodes = nodes && nodes.length;
-      // Don't show loading screen when we're auto-reloading
-      setShowLoading(!hasNodes);
-    }, [nodes]);
-
-    const handleLegendControlChange = useCallback(
-      (opts: LegendControlOptions) => {
-        changeBoundsOverride(opts.bounds);
-        changeAutoBounds(opts.auto);
-        changeLegend(opts.legend);
-      },
-      [changeBoundsOverride, changeAutoBounds, changeLegend]
-    );
-
-    return (
-      <>
-        <PageContent>
-          <EuiFlexGroup direction="column" gutterSize="s">
-            <TopActionContainer grow={false}>
-              <EuiFlexGroup justifyContent="spaceBetween" alignItems="center" gutterSize="m">
-                <Toolbar nodeType={nodeType} currentTime={currentTime} />
-                <EuiFlexGroup
-                  responsive={false}
-                  css={css`
-                    margin: 0;
-                    justify-content: flex-end;
-                  `}
-                >
-                  {view === 'map' && (
-                    <EuiFlexItem grow={false}>
-                      <LegendControls
-                        options={legend != null ? legend : DEFAULT_LEGEND}
-                        dataBounds={dataBounds}
-                        bounds={bounds}
-                        autoBounds={autoBounds}
-                        boundsOverride={boundsOverride}
-                        onChange={handleLegendControlChange}
-                      />
-                    </EuiFlexItem>
-                  )}
+    },
+    isAutoReloading ? 5000 : null
+  );
+
+  const dataBounds = calculateBoundsFromNodes(nodes);
+  const bounds = autoBounds ? dataBounds : boundsOverride;
+  /* eslint-disable-next-line react-hooks/exhaustive-deps */
```

Comment:
Q: We can maybe use useMemo here if we want to memoize the response to avoid disabling the eslint rule: 
Something like:
```
  const formatter = useMemo(
     createInventoryMetricFormatter(options.metric),
    [options.metric]
  );
```
maybe?

Reply:
I actually didn't change anything here as this code was just moved down, so I didn't check this part of the logic.
I'd rather keep the changes focused on the current task and apply refactors like this on another PR, does it sound good?

Reply:
Yes, it makes sense 👍